backend/pkg/api/data_access/vdb_helpers.go

```diff
@@ -400,3 +402,220 @@ func (d *DataAccessService) calculateValidatorDashboardBalance(ctx context.Conte
+		err = db.GetContext(ctx, &result, query, args...)
+	}
+
+	if err != nil {
+		return result, fmt.Errorf("error executing query: %w", err)
+	}
+
+	return result, nil
+}
+
+// --- Structs ---
+type SyncCommitteeResult struct {
+	ValidatorIndex uint64 `db:"validatorindex"`
+	Period         uint64 `db:"period"`
+}
+
+// --- Sync Committee Functions ---
+func (d *DataAccessService) getCurrentAndUpcomingSyncCommittees(ctx context.Context, latestEpoch uint64) (map[uint64]bool, map[uint64]bool, error) {
+	ds := buildSyncCommitteeQuery(latestEpoch)
+	queryResult, err := executeQuery[[]SyncCommitteeResult](ctx, d.readerDb, ds)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	current, upcoming := processSyncCommitteeResults(queryResult, utils.SyncPeriodOfEpoch(latestEpoch))
+	return current, upcoming, nil
+}
+
+func buildSyncCommitteeQuery(latestEpoch uint64) *goqu.SelectDataset {
+	currentSyncPeriod := utils.SyncPeriodOfEpoch(latestEpoch)
+	return goqu.Dialect("postgres").
+		Select(
+			goqu.L("validatorindex"),
+			goqu.L("period"),
+		).
+		From("sync_committees").
+		Where(goqu.L("period IN (?, ?)", currentSyncPeriod, currentSyncPeriod+1))
+}
+
+func processSyncCommitteeResults(queryResult []SyncCommitteeResult, currentSyncPeriod uint64) (map[uint64]bool, map[uint64]bool) {
+	currentSyncCommitteeValidators := make(map[uint64]bool)
+	upcomingSyncCommitteeValidators := make(map[uint64]bool)
+
+	for _, entry := range queryResult {
+		if entry.Period == currentSyncPeriod {
+			currentSyncCommitteeValidators[entry.ValidatorIndex] = true
+		} else {
+			upcomingSyncCommitteeValidators[entry.ValidatorIndex] = true
+		}
+	}
+
+	return currentSyncCommitteeValidators, upcomingSyncCommitteeValidators
+}
+
+// --- Epoch Start Functions ---
+func (d *DataAccessService) getEpochStart(ctx context.Context, period enums.TimePeriod) (uint64, error) {
+	ds, err := buildEpochStartQuery(d, period)
+	if err != nil {
+		return 0, err
+	}
+
+	epochStart, err := executeQuery[uint64](ctx, d.clickhouseReader, ds)
+	if err != nil {
+		return 0, err
+	}
+
+	return epochStart, nil
+}
+
+func buildEpochStartQuery(d *DataAccessService, period enums.TimePeriod) (*goqu.SelectDataset, error) {
+	clickhouseTable, _, err := d.getTablesForPeriod(period)
+	if err != nil {
+		return nil, err
+	}
+
+	ds := goqu.Dialect("postgres").
+		Select(goqu.L("epoch_start")).
+		From(goqu.L(fmt.Sprintf("%s FINAL", clickhouseTable))).
+		Order(goqu.L("epoch_start").Asc()).
+		Limit(1)
+
+	return ds, nil
+}
+
+// --- Past Sync Committees Functions ---
+func (d *DataAccessService) getPastSyncCommittees(ctx context.Context, indices []uint64, epochStart uint64, latestEpoch uint64) (map[uint64]uint64, error) {
+	ds := buildPastSyncCommitteesQuery(indices, epochStart, latestEpoch)
+	validatorIndices, err := executeQuery[[]uint64](ctx, d.alloyReader, ds)
+	if err != nil {
+		return nil, err
+	}
+
+	validatorCountMap := processPastSyncCommitteesResults(validatorIndices)
+	return validatorCountMap, nil
```

Comment:
Nitpick (non-blocking):
```suggestion
	return processPastSyncCommitteesResults(validatorIndices)
, nil
```

Reply:
Done